browser/src/canvas/sections/URLPopUpSection.ts

```diff
@@ -146,26 +146,27 @@ class URLPopUpSection extends HTMLObjectSection {
 	}
 
 	public static resetPosition(section: URLPopUpSection) {
-		if (!section)
-			section = app.sectionContainer.getSectionWithName(URLPopUpSection.sectionName);
+		if (!section) section = app.sectionContainer.getSectionWithName(URLPopUpSection.sectionName);
+		if (!section) return;
 
 		let left = section.sectionProperties.documentPosition.pX - section.getPopUpWidth() * 0.5 * app.dpiScale;
 		let top = section.sectionProperties.documentPosition.pY - (section.getPopUpHeight() + URLPopUpSection.popupVerticalMargin) * app.dpiScale;
 
-		if (section) {
-			let arrowAtTop = false;
-			if (top < 0) {
-				top = section.sectionProperties.documentPosition.pY + (URLPopUpSection.popupVerticalMargin * 2 * app.dpiScale);
-				arrowAtTop = true;
-			}
+		const checkLeft = (section.sectionProperties.documentPosition.pX - section.containerObject.getDocumentTopLeft()[0]) - section.getPopUpWidth() * 0.5 * app.dpiScale;
```

Comment:
isn't that just `checkLeft = left - section.containerObject.getDocumentTopLeft()[0])` ?

can we simplify?

Reply:
Yes but that is the check. Indeed that was above the "left "variable but i moved it below to make the review process easier (fewer changes). Let me send an update there. We can move cheks to top then we can use some maths there. It'll be better in terms of design i think.

Reply:
Okay done. Thank you for checking.